contracts/AMMs/PSMAMM.sol

```diff
@@ -18,7 +22,7 @@ contract PSMAMM is AMMBase {
         }
         psm = _psm;
         _psm.reserveToken().approve(address(_psm), type(uint256).max);
-        _psm.r().approve(address(_psm), type(uint256).max);
+        IPSMOld(address(_psm)).rToken().approve(address(_psm), type(uint256).max);
```

Comment:
I think this is a better solution than the current, because, if we deploy again PSMAMM contract (or a contract that inherits this), it'll not work with the current PSM deployed contract.